ecommerce/views/v0/__init__.py

```diff
@@ -425,110 +423,83 @@ def post(self, request, *args, **kwargs):
-                )
+                reverse("cart"), {"type": USER_MSG_TYPE_PAYMENT_DECLINED}
             )
+        elif order_state == Order.STATE.REVIEW:
+            basket = Basket.objects.filter(user=order.purchaser).first()
             if basket:
                 if basket.has_user_blocked_products(order.purchaser):
                     return redirect_with_user_message(
                         reverse("user-dashboard"),
                         {"type": USER_MSG_TYPE_ENROLL_BLOCKED},
                     )
                 else:
-                    basket.delete()
-
-            order.review(payment_data)
-            order.save()
-
-            if basket and basket.compare_to_order(order):
-                basket.delete()
-
+                    return redirect_with_user_message(
+                        reverse("user-dashboard"),
+                        {"type": USER_MSG_TYPE_PAYMENT_REVIEW},
+                    )
+        elif order_state == Order.STATE.FULFILLED:
             return redirect_with_user_message(
-                reverse("user-dashboard"), {"type": USER_MSG_TYPE_PAYMENT_REVIEW}
+                reverse("user-dashboard"),
+                {
+                    "type": USER_MSG_TYPE_PAYMENT_ACCEPTED,
+                    "run": order.lines.first().purchased_object.course.title,
+                },
             )
-        elif processor_response.state == ProcessorResponse.STATE_ACCEPTED:
-            # It actually worked here
-            self.logger.debug(
-                "Transaction accepted!: {msg}".format(msg=processor_response.message)
-            )
-
-            return api.fulfill_completed_order(order, payment_data, basket)
         else:
-            order.cancel()
-            order.save()
             return redirect_with_user_message(
                 reverse("user-dashboard"), {"type": USER_MSG_TYPE_PAYMENT_ERROR_UNKNOWN}
             )
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+class BackofficeCallbackView(APIView):
+    authentication_classes = []  # disables authentication
+    permission_classes = []  # disables permission
+
+    def post(self, request, *args, **kwargs):
+        """
+        This endpoint is called by Cybersource as a server-to-server call
+        in order to respond with the payment details.
+        """
+        order = api.get_order_from_cybersource_payment_response(request)
+
+        # We only want to process responses related to orders which are PENDING
+        # otherwise we can conclude that we already received a response through
+        # the user's browser.
+        result_status = status.HTTP_200_OK
+        try:
+            if order is not None and order.state == Order.STATE.PENDING:
+                api.process_cybersource_payment_response(request, order)
+            elif order is None:
+                raise ObjectDoesNotExist()
+        except ObjectDoesNotExist:
+            result_status = status.HTTP_404_NOT_FOUND
```

Comment:
I think you can simplify this down to avoid the `try...except` - it adds unnecessary complexity.

Reply:
@rhysyngsun Should I just return the `404` and not raise the exception, or is there a way in which both can be done?